source/_posts/2016-01-25-upgrading-to-ruby-on-rails-5-0.markdown

```diff
@@ -0,0 +1,226 @@
+
+
+## First naive attempt to upgrade
+
+Given that our application already worked with Ruby 2.2.2, we didn’t have to do anything in this respect.
+
+The first step we took was to modify the Gemfile by changing the Rails version from 4.2.5 to 5.0.0.beta1. The Gemfile needs to have a line with `gem 'rails', '5.0.0.beta1'`.
+
+The following step is to execute `bundle update rails` and ...
+
+```
+    devise was resolved to 3.5.2, which depends on
+      railties (< 5, >= 3.2.6)
+```
+
+It was a very likely possibility that some gems were still not compatible with Rails 5. In the case of devise, the master version does indeed support Rails 5 and we therefore change the devise line in the `Gemfile` to `gem 'devise', github: 'plataformatec/devise'`.
+
+We again execute `bundle update rails` and ...
+
+```
+    minitest-rails was resolved to 2.2.0, which depends on
+      railties (~> 4.1)
+```
+
+For this gem, there’s no version that will fix the problem. Still, the first thing I asked myself was, what things were we using from that gem. The gem sets up the minitest spec DSL and adds some generators, but we weren’t using the DSL, nor the generators, so we simply erase this Gemfile entry. 
+
+We execute `bundle update rails` again ...
+
+```
+    minitest-rails-capybara was resolved to 2.1.1, which depends on
+      minitest-rails (~> 2.1) was resolved to 2.2.0, which depends on
+        railties (~> 4.1)
+```
+
+We are using minitest-rails-capybara, which depends on minitest-rails. We just erased the Gemfile, but owing to this dependency, minitest-rails continues trying to install. We follow the same procedure from the previous case, asking ourselves, “What do we need minitest-rails-capybara for?” The reality is that we don’t need it either, we’ll simply use minitest-capybara, which is one of its dependencies.
+
+Therefore:
+
+- We substitute the minitest-capybara for the minitest-rails-capybara entry
+- We call a require of `capybara/rails` in place of `minitest/rails/capybara` in the `test_helper.rb`.
+- We make sure that the integration tests inherit from `ActionDispatch::IntegrationTest` instead of from `Capybara::Rails::TestCase` (a class of minitest-rails-capybara).
+- And lastly we add the following to `test_helper.rb` ...
+
+```
+class ActionDispatch::IntegrationTest
+  include Capybara::DSL
+  include Capybara::Assertions
+
+  ...
+end
+```
+
+We execute `bundle update rails` again and now it installs everything correctly.
+
+Now we can run the tests, for example.
+Hence, we execute `bin/rails test` (this is the new way to run tests in Rails 5!) and we have various tests failing.
+
+
+## Problem 1 - the tests fail and we don’t have any further information
+
+From this point forward, the Rails guides give us no further information and we still have various tests failing. In particular, failing is a series of tests of the following form ...
```

Comment:
`failing is a series of tests of the following form` => `there are some failing tests with the following form`